### ISSUE TITLE
Invalid SmartThinQ credentials error

### ISSUE THREAD

**Issue by pbix** [ollo69/ha-smartthinq-sensors#218]
I have been using this integration for some time and it worked well.  I got the error that my credentials had expired and am trying to correct that.  

Things I have done.
1) Accepted the new TOS using my Thinq app on my phone.  My phone app works well by the way, no issue noticed.
2) Deleted, upgraded, rebooted and reinstalled the custom_component.
3) Went through the UI and proceed to obtain the redirect URL.  That process requires login and seems to proceed as normal.  I know I am logged in because if I use a random password I get a completely different screen (not the redirect URL).
4) Copy-Paste the URL and get the below error:

![image](https://user-images.githubusercontent.com/289729/135659126-13b2f56e-894d-4e05-aa11-25a14300e812.png)

Any help would be appreciated.

**Comment by ollo69** [comment]
Dir you try checking the 2 options below (may be one at time)?

**Comment by pbix** [comment]
I did try the first one one time.  I never tried the second one because I did not see the 'dh key too small' error.

**Comment by miklosandras** [comment]
I have the same issue with v0.8.20 and v0.10.0

"Invalid SmartThinQ credentials. Try again, and make sure you can access the ThinQ app on your phone."

Also tried both option.

**Comment by miklosandras** [comment]
I have the same issue with v0.8.20 and v0.10.0

"Invalid SmartThinQ credentials. Try again, and make sure you can access the ThinQ app on your phone."

Also tried both option.

**Comment by Njord85** [comment]
I have the same issue since the last beta update

**Comment by miklosandras** [comment]
@ollo69 said:

"When problem is related to TOS change's, the solution is not re-configure integration but just accept TOS on phone App and rebooting HA everything works properly again."

It is worked, after I accepted in mobile app, I can added into HA.

**Comment by pbix** [comment]
I have rebooted twice and not luck. Same message as shown above.  Very frustrating.

**Comment by foskasse** [comment]
Same Issue here

**Comment by ollo69** [comment]
I have no idea why some people having this issue. On my side I just accept TOS on the phone, restarted HA and connection come back again. Are you sure that you are using same account on LG app and integration?
If you don't solve the issue in any way, enable debug log and attach it here, I will analyze it to see if I can found the cause of the error.

**Comment by pbix** [comment]
For the benefits of others that read this thread I will detail what happened to me.  I unfortunately have both an LG account created on my phone app using "signin with Google" AND an account using a yahoo.com email over the lg.com website.  Make sure to look at your profile on your phone app and see its the same email you are using to create the token URL.  That whole process works with the "signin with google" stuff.  Contrary to what I have seen written you can use this account both in the phone app and the HA integration.  What I have not be able to see is how to sign in at lg.com using the google linked account.  There is no option today to "sign in with google" there.

All I can advise if you have this issue is to keep good notes.  For warranty work with LG you need to sign into the website.  If anyone can tell me how to signin with google when there is no option to do that on the web page I would appreciate it.

I cannot find any issue with the excellent integration that ollo69 has provided us.  You just have to be clear on the accounts you have and which is being used for what.

So in the end I was confused by all this security forced on use users.  Frustrating because passwords are n fact most often compromised by hacks into these companies' databases and not be some Russian trying to guess our passwords.  Yet they continue to make signing in for legit users harder every year.

**Comment by BillyFKidney** [comment]
Is there any way to have HA to show an alert for the integration when the ToS change?

I don’t use the LG app and didn’t know the ToS changed until I found this thread?


> @ollo69 said:
> 
> "When problem is related to TOS change's, the solution is not re-configure integration but just accept TOS on phone App and rebooting HA everything works properly again."
> 
> It is worked, after I accepted in mobile app, I can added into HA.

**Comment by ollo69** [comment]
> Is there any way to have HA to show an alert for the integration when the ToS change?

When ToS change, API only return credential error, there are no way to retrieve more details. I could add a notification in HA when this error is returned, but nothing more.

**Comment by tismofied** [comment]
Are there any issues on version 0.10.0 or should I stay on version 0.9?￼

**Comment by TheGabeMan** [comment]
For info: Accepting the TOS in the app and then rebooting HA solved it for me.

**Comment by felixfischer** [comment]
That also solved it for me ✔️

**Comment by BillyFKidney** [comment]
@ollo69 
> > Is there any way to have HA to show an alert for the integration when the ToS change?
> 
> When ToS change, API only return credential error, there are no way to retrieve more details. I could add a notification in HA when this error is returned, but nothing more.

That would be a huge help and probably result in less bug reports when this eventually happens again

...something to the effect of "Credential Error returned, please use the LG.app to ensure credentials are correct then restart HA" would result in users accepting the new ToS.

**Comment by jeffgoh** [comment]
> I have no idea why some people having this issue. On my side I just accept TOS on the phone, restarted HA and connection come back again. 

It only just started happening for me in Singapore a day ago (they must be doing a few regions at a time) - all I had to do was accept TOS on the LG App - apparently they block logins until then.  Once I restarted HA, all good. Thanks again for an amazing integration , @ollo69 - being able to say "Hey Google, turn off <room-name> aircon" via my Home Mini and actually have it work consistently is a life changer.  Definitely beats searching for the remote the wife has "misplaced" :)
